intake_esm/source.py

```diff
@@ -73,6 +73,9 @@ def _open_dataset(
             requested_variables = [requested_variables]
         variable_intersection = set(requested_variables).intersection(set(varname))
         variables = [variable for variable in variable_intersection if variable in ds.data_vars]
+        variables.extend(
+            {ds[v].attrs['grid_mapping'] for v in variables if 'grid_mapping' in ds[v].attrs}
+        )
```

Comment:
@RondeauG, how about we use your suggestion in https://github.com/intake/intake-esm/pull/449/files#r833298541?

```suggestion
       scalar_variables = [v for v in ds.data_vars if len(ds[v].dims) == 0]
       ds = ds.set_coords(scalar_variables)
```

this should cover all the issues (including the concatenation, etc...)

Reply:
once this is taken care of, this PR should be good to be merged